src/core/reducers/versions.js

```diff
@@ -100,14 +102,43 @@ export const createInternalVersion = (
       version.is_strict_compatibility_enabled,
     ),
     license: version.license
-      ? { name: version.license.name, url: version.license.url }
+      ? {
+          name: version.license.name,
+          text: version.license.text,
+          url: version.license.url,
+        }
       : null,
     platformFiles: createPlatformFiles(version),
     releaseNotes: version.release_notes,
     version: version.version,
   };
 };
 
+type FetchVersionParams = {|
+  errorHandlerId: string,
+  slug: string,
+  versionId: VersionIdType,
+|};
+
+export type FetchVersionAction = {|
+  type: typeof FETCH_VERSION,
+  payload: FetchVersionParams,
+|};
+
+export const fetchVersion = ({
+  errorHandlerId,
+  slug,
+  versionId,
+}: FetchVersionParams): FetchVersionAction => {
+  invariant(errorHandlerId, 'errorHandlerId is required');
+  invariant(slug, 'slug is required');
```

Comment:
We need an `invariant` call for `versionId` too, no?

Reply:
I'm actually kind of confused about when we need `invariant` and when we don't. This is covered by Flow, so shouldn't that protect us and mean we don't need any of these `invariant`s? It seems like we often don't use `invariant` for required arguments because we have Flow, so why do we need it here? Is it because not all of our code is covered by Flow? If that's the case, how do we decide when we need `invariant`s and when we don't?

Reply:
All action creators need `invariant` because a lot of components are not covered by Flow, and action creators are used in the component files. The rest should not need `invariant` because we have better coverage (it is easier to convert an API file to flow than a React component).

We also use `invariant` in the React components when Flow is a bit lost (because of the default props or because a function is only called in a specific context).